### PR TITLE
📖 Document backporting rules for Go lang minor versions for images

### PR DESCRIPTION
**Pull request by kashifest**
This PR also rewrites the Backporting section a bit and puts on clear identifiers for backporting rules.

### CODE REVIEW

**Comment by kashifest** [comment]
@dtantsur @honza @elfosardo please review this before it rots

**Comment by kashifest** [comment]
are we going to get any eyes on this?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -84,10 +84,34 @@ Types and functions not in public APIs are not considered part of the guarantee.
...
 #### Backporting
 
-We only accept backports of critical bugs, security issues, or bugs without
-easy workarounds, any
-backport MUST not be breaking for either API or behavioral changes.
-We generally do not accept PRs against older release branches.
+We generally do not accept PRs directly against release branches, while we might
+accept backports of fixes/changes already merged into the main branch.
+
+We generally allow backports of following changes to all supported branches:
```
Comment (honza) [inline CONTRIBUTING.md:90]: "We generally allow backports of the following kinds of changes to all supported branches:"

Comment (zaneb) [reply CONTRIBUTING.md]: all _maintained_ branches

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -84,10 +84,34 @@ Types and functions not in public APIs are not considered part of the guarantee.
...
+- Improvements to existing docs
+- Improvements to the test framework
+
+Like any other activity in the project, backporting a fix/change is a
+community-driven effort and requires that someone volunteers to own the task.
+In most cases, the cherry-pick bot can (and should) be used to automate
+opening a cherry-pick PR.
+
+We generally do not accept backports to BMO release branches that are out of
+support. Check the [Version support](https://github.com/metal3-io/metal3-docs/blob/main/docs/user-guide/src/version_support.md#baremetal-operator)
```
Comment (zaneb) [inline CONTRIBUTING.md:114]: s/out of support/at end of life/

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -84,10 +84,35 @@ Types and functions not in public APIs are not considered part of the guarantee.
...
 #### Backporting
 
-We only accept backports of critical bugs, security issues, or bugs without
-easy workarounds, any
-backport MUST not be breaking for either API or behavioral changes.
-We generally do not accept PRs against older release branches.
+We generally do not accept PRs directly against release branches, while we might
+accept backports of fixes/changes already merged into the main branch.
+
+We generally allow backports of the following kinds of changes to all supported
```
Comment (zaneb) [inline CONTRIBUTING.md:90]: all _maintained_ branches

File: CONTRIBUTING.md (hunk 1 above)
Comment (kashifest) [reply CONTRIBUTING.md]: Thanks done

**Review by zaneb — COMMENTED** [review]
Open source communities don't provide support, but we do maintain stuff.

File: CONTRIBUTING.md (hunk 2 above)
Comment (kashifest) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 3 above)
Comment (kashifest) [reply CONTRIBUTING.md]: done

**Review by tuminoid — COMMENTED** [review]
/lgtm

**Pull request merged**